infrastructure/container/container.go

```diff
@@ -0,0 +1,331 @@
+/*
+ * © 2025 Snyk Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package container
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"regexp"
+	"strings"
+
+	"github.com/rs/zerolog/log"
+
+	"github.com/snyk/snyk-ls/application/config"
+	"github.com/snyk/snyk-ls/domain/snyk"
+	"github.com/snyk/snyk-ls/internal/observability/error_reporting"
+	"github.com/snyk/snyk-ls/internal/product"
+	"github.com/snyk/snyk-ls/internal/types"
+)
+
+const (
+	containerDocumentationUrl = "https://docs.snyk.io/products/snyk-container"
+)
+
+type Scanner struct {
+	config        *config.Config
+	errorReporter error_reporting.ErrorReporter
+	docUrl        *url.URL
+	httpClient    func() *http.Client
+	apiClient     BaseImageRemediationAPI
+}
+
+// BaseImageRemediationAPI defines the interface for base image remediation API
+type BaseImageRemediationAPI interface {
+	GetBaseImageRecommendation(ctx context.Context, baseImage string) (*BaseImageRecommendation, error)
+}
+
+// BaseImageRecommendation represents a recommended base image upgrade
+type BaseImageRecommendation struct {
+	OriginalImage     string
+	RecommendedImage  string
+	Reason            string
+	SeverityReduction string
+}
+
+// DockerDepsResponse represents the response from docker-deps recommended-base-image endpoint
+type DockerDepsResponse struct {
+	BaseImage         string                `json:"baseImage"`
+	Recommendations   []ImageRecommendation `json:"recommendations"`
+	MinorUpgrades     []ImageRecommendation `json:"minorUpgrades"`
+	MajorUpgrades     []ImageRecommendation `json:"majorUpgrades"`
+	AlternativeImages []ImageRecommendation `json:"alternativeImages"`
+}
+
+// ImageRecommendation represents a single image recommendation
+type ImageRecommendation struct {
+	Image  string `json:"image"`
+	Reason string `json:"reason"`
+}
+type DockerDepsRemediationAPI struct {
+	httpClient func() *http.Client
+	config     *config.Config
+}
+
+func (d *DockerDepsRemediationAPI) GetBaseImageRecommendation(ctx context.Context, baseImage string) (*BaseImageRecommendation, error) {
+	// Construct the API URL
+	apiURL := fmt.Sprintf("%s/docker-deps/recommended-base-image", d.config.SnykApi())
+
+	// Create the request
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, apiURL, nil)
```

Comment:
As container scanning is a long running process (I think), is there a way to report progress e.g. polling progress from the API?